src/ipa_pc/mod.rs

```diff
@@ -659,7 +667,9 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
 
         let combine_time = start_timer!(|| "Combining polynomials, randomness, and commitments.");
 
-        let mut cur_challenge: G::ScalarField = fs_rng.squeeze_128_bits_challenge();
+        // TODO: we need to compute the values and absorb them before squeezing the challenge
```

Comment:
In Marlin the evaluations are already absorbed (https://github.com/HorizenLabs/marlin/blob/dev/src/lib.rs#L279), so we shouldn't need to do anything. We can just add a disclaimer that the fs_rng should already have absorbed the evaluations values.
It's true that the evaluations are recomputed also here (and it's useless), maybe we can open an issue that in the future we pass also the evaluations and we absorb them here and we avoid recomputing them.

Reply:
Ok, that is also fine. I'll change the comments accordingly.

Reply:
done